main.go

```diff
@@ -812,6 +812,35 @@ func (task *TaskRun) ExecuteCommand(index int) *CommandExecutionError {
 	if err != nil {
 		return WorkerShutdown(err)
 	}
+
+	// the duration until the expiry is precisely -1 *
+	// the amount of time since the expiry
+	duration := -time.Since(time.Time(task.Definition.Expires))
+	commandLogURL, err := Queue.GetArtifact_SignedURL(
```

Comment:
If you don't have required scopes to read the livelog artifact, you won't be able to get the signed URL. If you've retrieved the signed URL, then you already had scopes to retrieve the livelog URL, so you should be allowed access to the s3 url.

But agreed if you leave it with the private artifact path, the person would need to log in, so that still works. I hadn't considered that. So I'll not sign the URL, and keep it simple. Thanks for highlighting a much simpler solution! :+1: